tests/mock/catalog.rs

```diff
@@ -29,7 +29,11 @@ fn test_catalog_simple() {
 
     let futcheck = dclient.get_catalog(None);
 
-    let res = runtime.block_on(futcheck.collect()).unwrap();
+    let (res, _errors): (Vec<_>, Vec<_>) = runtime
+        .block_on(futcheck.collect::<Vec<_>>())
+        .into_iter()
+        .partition(Result::is_ok);
+    let res: Vec<_> = res.into_iter().map(Result::unwrap).collect();
```

Comment:
This should likely also assert that `errors.len() == 0`.

Reply:
In that case we can probably just omit the partitioning.